src/dispatch-payload.ts

```diff
@@ -30,13 +27,25 @@ type GetFunctionFileCallback = {
   (functionCallbackId: string): string | FunctionModule;
 };
 
+// A helper structure that maps event names to the functions responsible for invoking userland code with the relevant event's payload
+const EVENT_TO_HANDLER_MAP = {
+  [EventTypes.FUNCTION_EXECUTED]: RunFunction,
+  [EventTypes.BLOCK_ACTIONS]: RunBlockAction,
+  [EventTypes.BLOCK_SUGGESTION]: RunBlockSuggestion,
+  [EventTypes.VIEW_CLOSED]: RunViewClosed,
+  [EventTypes.VIEW_SUBMISSION]: RunViewSubmission,
+};
+
 export const DispatchPayload = async (
   // deno-lint-ignore no-explicit-any
   payload: InvocationPayload<any>,
   hookCLI: Protocol,
   getFunctionFile: GetFunctionFileCallback,
 ) => {
-  const eventType = payload?.body?.event?.type || payload?.body?.type || "";
+  // TODO: should we check that this is a ValidEventType at runtime?
```

Comment:
This may not be a relevant TODO, I do recall discussion about not wanting to enforce the event type too much in order to be able to experiment with new event types.